applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.cpp

```diff
@@ -0,0 +1,476 @@
+            noalias(r_integrated_stress_vector) = (1.0 - damage) * predictive_stress_vector;
+
+            if (r_constitutive_law_options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+                r_constitutive_matrix *= (1.0 - damage);
+                this->SetStressVector(r_integrated_stress_vector);
+            }
+        } else { // Damage case
+            const double characteristic_length = ConstitutiveLawUtilities<VoigtSize>::CalculateCharacteristicLength(rValues.GetElementGeometry());
+            //const double characteristic_length =0.01;
+            // This routine updates the PredictiveStress to verify the yield surf
+            TConstLawIntegratorType::IntegrateStressVector(
+                predictive_stress_vector, 
+                uniaxial_stress, 
+                damage, 
+                threshold, 
+                rValues, 
+                characteristic_length);
+
+            // Updated Values
+            noalias(r_integrated_stress_vector) = predictive_stress_vector;
+            
+            if (r_constitutive_law_options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+                TConstLawIntegratorType::YieldSurfaceType::CalculateEquivalentStress(predictive_stress_vector, r_strain_vector,uniaxial_stress, rValues);
+                this->SetStressVector(r_integrated_stress_vector);
+                this->CalculateTangentTensor(rValues);
+            }
+        }
+    }
+}
+
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+void GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    // Integrate Stress Damage
+    const Flags& r_constitutive_law_options = rValues.GetOptions();
+    Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();
+
+    // We get the strain vector
+    Vector& r_strain_vector = rValues.GetStrainVector();
+
+    //NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
+    if (r_constitutive_law_options.IsNot( ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
+        this->CalculateValue(rValues, STRAIN, r_strain_vector);
+    }
+
+    // Elastic Matrix
+    if (r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+        this->CalculateValue(rValues, CONSTITUTIVE_MATRIX, r_constitutive_matrix);
+    }
+    array_1d<double, VoigtSize> predictive_stress_vector;
+    // We compute the stress
+    if(r_constitutive_law_options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+        // Elastic Matrix
+        this->CalculateValue(rValues, CONSTITUTIVE_MATRIX, r_constitutive_matrix);
+
+        // Converged values
+        double threshold = this->GetThreshold();
+        double damage = this->GetDamage();
```

Comment:
const

Reply:
I can not set them into const. They are recalculated in the "TConstLawIntegratorType::IntegrateStressVector ( )"